Generated by startproject, to interact with pj

XiuqinCandice opened the pull request:
Dear@listonjesse,

I created the 1st project by following this [link](https://docs.djangoproject.com/en/3.0/intro/tutorial01/)

XiuqinCandice commented:
👍 Sure, to be specific, Did you mean change the directory 'polls' to 'todo'?

listonjesse commented:
> 👍 Sure, to be specific, Did you mean change the directory 'polls' to 'todo'?

I think the you should run 'python manage.py startapp todo' and delete the polls the directory. Renaming the directory would not be enough because there are files generated that contain the name [poll](https://github.com/XiuqinCandice/to-do-webapp/blob/1st-Django-app/basic_poll/mysite/polls/apps.py)

In addition to redoing the 'startapp' part, I think you should also rename the top directory from 'basic_poll' to 'mysite'

So in the end the directory structure will look like this:
mysite/
    manage.py
    mysite/
        __init__.py
        settings.py
        urls.py
        asgi.py
        wsgi.py
    todo/
        __init__.py
       admin.py
       apps.py
       migrations/
           __init__.py
       models.py
       tests.py
       views.py

XiuqinCandice commented:
Thank you for the comments, I've done the following steps

- Moved the outer mysite folder to the parent level (basic_poll) and delete basic_poll.

- Restarted a new app to do and deleted polls

- Pointed the root URLconf at the todo.urls module
Can you help double-check the directory structure? Thanks a lot

listonjesse commented on mysite/todo/urls.py:
```diff
@@ -0,0 +1,7 @@
+from django.urls import path
+
```
nitpick: remove this new line

XiuqinCandice replied:
Thanks

listonjesse commented on mysite/todo/views.py:
```diff
@@ -0,0 +1,5 @@
+from django.http import HttpResponse
+
```
nitpick: put just one line between the imports and the first function definition

XiuqinCandice replied:
Thank you

XiuqinCandice commented:
Thank you @listonjesse I couldn't find .DS_Store file in the directory but I've added .DS_Store to .gitignore

listonjesse commented:
@XiuqinCandice The .DS_Store file is a hidden file on mac, so you will need to run 'git rm .DS_Store' in a terminal to remove it since it won't show up in VSCode. Then when you remove it locally, you can commit the deletion

listonjesse commented:
![image](https://user-images.githubusercontent.com/7820952/75076104-20d68e80-54b4-11ea-9e90-0a51dec04ad9.png)

Don't forget to link the issues this PR solves (i.e. both Do Django tutorial and v1 web app issues) and also link our project so the board will update

Thanks!

XiuqinCandice commented:
> @XiuqinCandice The .DS_Store file is a hidden file on mac, so you will need to run 'git rm .DS_Store' in a terminal to remove it since it won't show up in VSCode. Then when you remove it locally, you can commit the deletion

Thanks for your help @listonjesse . I've successful removed the file and committed it

XiuqinCandice commented:
> ![image](https://user-images.githubusercontent.com/7820952/75076104-20d68e80-54b4-11ea-9e90-0a51dec04ad9.png)
> 
> Don't forget to link the issues this PR solves (i.e. both Do Django tutorial and v1 web app issues) and also link our project so the board will update
> 
> Thanks!

Do you mean this issue: "Set up repo with bare minimum web app using Django (Hello World) #1"?. I've linked it with this pull request, let me know. Thanks

listonjesse commented:
> > ![image](https://user-images.githubusercontent.com/7820952/75076104-20d68e80-54b4-11ea-9e90-0a51dec04ad9.png)
> > Don't forget to link the issues this PR solves (i.e. both Do Django tutorial and v1 web app issues) and also link our project so the board will update
> > Thanks!
> 
> Do you mean this issue: "Set up repo with bare minimum web app using Django (Hello World) #1"?. I've linked it with this pull request, let me know. Thanks

Yes, looks good to me now. Thanks!

listonjesse commented:
@XiuqinCandice Can you make sure to "Squash and Merge"? It will make our repo history look a lot tidier

![image](https://user-images.githubusercontent.com/7820952/75081956-31900000-54c6-11ea-9f63-d26f87bfe5d4.png)

XiuqinCandice commented:
👍 💯 Will do, thank you @listonjesse.

XiuqinCandice commented:
I found that even though I delete the branch, I can still see them in local, do I need to git branch -d 'branchname' to delete it and do git fetch -p to look for dead branches and remove those references at remote
